API: Compute schema once and share among other nodes

joselcvarela opened the pull request:
## Scope
In some scenarios we see the slow starts along with `Failed to get Schema information: hit infinite loop` and `Couldn't load extensions` messages.
This seems to be caused by the way the schema is fetched.
Recently, [on this PR](https://github.com/directus/directus/pull/21824), there was made a change to compute the schema only once.
Although, the schema is stored in memory so it means that all the nodes will require to compute the cache again.

In this PR, we just take advantage of just one node compute the cache and share it with other nodes via redis pub/sub.

What's changed:

- Compute schema once and share among other nodes

## Potential Risks / Drawbacks

- For large schemas, Redis bandwidth should be taken into consideration

## Review Notes / Questions

- This branch was made from `v10.13.2`, so it can be used to deploy a v10 release if needed

joselcvarela committed:
Compute schema once and share among other nodes

rijkvanzanten commented on api/src/utils/get-schema.ts:
```diff
@@ -68,41 +68,37 @@ export async function getSchema(
 	} finally {
 		await lock.delete(lockKey);
-		bus.publish(messageKey, { ready: true });
+		bus.publish(messageKey, { ready: true, schema });
```
`schema` here can be null

rijkvanzanten committed:
Add compress/decompress

rijkvanzanten committed:
Remove weirdness

rijkvanzanten committed:
Make timeout configurable

joselcvarela committed:
Remove compression/decompression
This was putting back the issues again.
I guess it's because the transformation to async
It seems useBus is already doing compression
So I believe there's no need for double compression.

rijkvanzanten committed:
Add changeset
